appserver/tests/embedded/web/servlet/src/test/java/org/glassfish/tests/embedded/web/servlet/Application.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ * Copyright (c) 2009, 2018 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.tests.embedded.web.servlet;
+
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+
+import org.glassfish.embeddable.GlassFish;
+import org.glassfish.embeddable.GlassFishException;
+import org.glassfish.embeddable.GlassFishProperties;
+import org.glassfish.embeddable.GlassFishRuntime;
+import org.glassfish.embeddable.archive.ScatteredArchive;
+
+
+public class Application implements Closeable {
+
+    private final GlassFish glassfish;
+    private final int port;
+    private final String name;
+
+    private Application(final GlassFish glassfish, final int port, final String name) {
+        this.glassfish = glassfish;
+        this.port = port;
+        this.name = name;
+    }
+
+
+    public URL getEndpoint() {
+        try {
+            return new URI("http://localhost:" + port + "/" + name + "/hello").toURL();
+        } catch (MalformedURLException | URISyntaxException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+
+    @Override
+    public void close() throws IOException {
+        try {
+            System.out.println("Undeploying");
+            glassfish.getDeployer().undeploy(name);
+            System.out.println("Stopping the server !");
+            glassfish.dispose();
+        } catch (final GlassFishException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+
+    public static Application start() throws IOException, GlassFishException {
+        GlassFishProperties props = new GlassFishProperties();
+        props.setPort("http-listener", 8882);
```

Comment:
Could the port be passed to `start`? The parameter could be reused in l. 84 if `8882` there represents the same.

Reply:
I was so happy I fixed that so I forgot, thanks! We can even use any random free port, I pushed that right now.